### PR TITLE
Fixes #1146 - Duplicate role creation in Export-SqlUser

### LINKED ISSUE #1146: When exporting database users via Export-SqlUser role creation happens multiple times
## Is this:
_Just place "x" to indicate selection (e.g. "[x]")_

- [ ] *feature* 
- [ X ] *bug*:

#### Example:
1. Use Export-SqlUser on a database with more than one user assigned to each role 

## Action Results
Your output will repeat the role creation for each user which has this role.

#### Example:
Your output should only create the role one time. 

From:
https://sqlcommunity.slackarchive.io/dbatools/page-100/ts-1494166741688394
User torustad reported that when exporting users that role creation was repeated for each user.

### PR DESCRIPTION
**Pull request by ConstantineK**
Fixes # 

Changes proposed in this pull request:
 - Create a list and track the roles created at export time, skipping it if it already exists.

How to test this code: 
- Run Export-SqlLogin on a box with a multiple users assigned to the same custom role.

Has been tested on minimum requirements:
- This change should work on all versions Export-SqlLogin works on  

Has been tested on maximum requirements:
- This change should work on all versions Export-SqlLogin works on 

Tests for tester:
- No additional tests have been added.

### CODE REVIEW

**Comment by potatoqualitee** [comment]
aw fuck, there's a conflict. @ConstantineK - can you check this out>

**Comment by wsmelton** [comment]
Likely from the PR merged where this file was renamed?

**Comment by wsmelton** [comment]
Yeah, this PR got merged that touched this file as well: #1203

**Comment by ConstantineK** [comment]
Sure, let me see if I can resolve the merge conflict by updating my branch and merging the changes.

File: functions/Export-DbaUser.ps1 [hunk 1]
```diff
@@ -202,21 +202,26 @@ https://dbatools.io/Export-DbaUser
                         $users = $db.Users | Where-Object { $User -contains $_.Name -and $_.IsSystemObject -eq $false -and $_.Name -notlike "##*" }
                     }
                 }
-				
+                # Store roles between users so if we hit the same one we dont create it again 
+                $roles = @()
```
Comment (ConstantineK) [inline functions/Export-DbaUser.ps1]: This is where my non-merge changes begin and end.

File: functions/Export-DbaUser.ps1 [hunk 2]
```diff
@@ -89,7 +89,7 @@ https://dbatools.io/Export-DbaUser
     param (
         [parameter(Mandatory = $true, ValueFromPipeline = $true)]
         [Alias("ServerInstance", "SqlServer")]
-        [object]$SqlInstance,
+        [DbaInstanceParameter]$SqlInstance,
```
Comment (ConstantineK) [inline functions/Export-DbaUser.ps1]: Did we change SqlInstance back from DbaInstanceParameter to object?

Comment (potatoqualitee) [inline functions/Export-DbaUser.ps1]: both are valid

**Pull request merged**